openssl/src/cms.rs

```diff
@@ -6,16 +6,59 @@
 //! Data accepted by this module will be smime type `enveloped-data`.
 
 use ffi;
-use foreign_types::{ForeignType, ForeignTypeRef};
+use foreign_types::{ForeignType, ForeignTypeRef, Opaque};
+use libc::c_void;
 use std::ptr;
 
+use asn1::Asn1Object;
 use bio::{MemBio, MemBioSlice};
 use error::ErrorStack;
+use hash::MessageDigest;
 use libc::c_uint;
 use pkey::{HasPrivate, PKeyRef};
-use stack::StackRef;
-use x509::{X509, X509Ref};
-use {cvt, cvt_p};
+use stack::{Stack, StackRef};
+use x509::store::X509Store;
+use x509::{X509Ref, X509};
+use {cvt, cvt_n, cvt_p};
+
+pub struct CmsSignerInfoRef(Opaque);
```

Comment:
You should be able to use the macro to create these.

Reply:
Sorry, but I'm unable to figure out the correct macro for this.